### PR TITLE
*: add testSuite4 back and rewrite TestBatchInsertDelete

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
testSuite4 is ignored for a long time, but add it back will cause some tests to fail. 

### What is changed and how it works?
1. add testSuite4 back
1. rewrite TestBatchInsertDelete to make it not happen in the future.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: executor/write_test.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func (s *testSuite4) TestInsert(c *C) {
 	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1690 constant -1 overflows bigint"))
 	tk.MustExec("insert into t select cast(-1 as unsigned);")
 	tk.MustExec("insert into t value (-1.111);")
-	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1690 constant -1 overflows bigint"))
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1690 constant -1.111 overflows bigint"))
```
Comment (jackysp) [inline executor/write_test.go]: Either -1 or -1.111 is not the same as MySQL.

**Comment by jackysp** [comment]
PTAL @eurekaka

File: executor/write_test.go [hunk 2]
```diff
@@ -2040,119 +2040,91 @@ func (s *testSuite4) TestLoadDataOverflowBigintUnsigned(c *C) {
 
 func (s *testSuite4) TestBatchInsertDelete(c *C) {
 	originLimit := atomic.LoadUint64(&kv.TxnEntryCountLimit)
-	defer func() {
-		atomic.StoreUint64(&kv.TxnEntryCountLimit, originLimit)
-	}()
 	// Set the limitation to a small value, make it easier to reach the limitation.
-	atomic.StoreUint64(&kv.TxnEntryCountLimit, 100)
+	var limit4Test uint64 = 2000
+	atomic.StoreUint64(&kv.TxnEntryCountLimit, limit4Test)
```
Comment (eurekaka) [inline executor/write_test.go]: Would this cause unexpected results for other tests running in parallel?

Comment (jackysp) [reply executor/write_test.go]: There may still be an impact on other tests, but
1. this value will be set back soon now
1. the value itself has been changed to 2000, so it is not easy to reach its limit in other cases.

Comment (eurekaka) [reply executor/write_test.go]: Or we can put this test case into executor/seqtest to run it sequentially?

**Comment by jackysp** [comment]
closed with https://github.com/pingcap/tidb/pull/10602/files

**Pull request closed without merging**